styx-api-service/src/main/java/com/spotify/styx/api/StyxConfigResource.java

```diff
@@ -1,104 +0,0 @@
-/*-
- * -\-\-
- * Spotify Styx API Service
- * --
- * Copyright (C) 2016 Spotify AB
- * --
- * Licensed under the Apache License, Version 2.0 (the "License");
- * you may not use this file except in compliance with the License.
- * You may obtain a copy of the License at
- * 
- *      http://www.apache.org/licenses/LICENSE-2.0
- * 
- * Unless required by applicable law or agreed to in writing, software
- * distributed under the License is distributed on an "AS IS" BASIS,
- * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- * See the License for the specific language governing permissions and
- * limitations under the License.
- * -/-/-
- */
-
-package com.spotify.styx.api;
-
-import static com.spotify.styx.api.Api.Version.V3;
-import static com.spotify.styx.api.Middlewares.json;
-
-import com.fasterxml.jackson.annotation.JsonProperty;
-import com.google.auto.value.AutoValue;
-import com.google.common.base.Throwables;
-import com.spotify.apollo.Request;
-import com.spotify.apollo.Response;
-import com.spotify.apollo.Status;
-import com.spotify.apollo.route.AsyncHandler;
-import com.spotify.apollo.route.Route;
-import com.spotify.styx.storage.Storage;
-import java.io.IOException;
-import java.util.Arrays;
-import java.util.List;
-import java.util.Objects;
-import java.util.Optional;
-import java.util.stream.Stream;
-import okio.ByteString;
-
-public class StyxConfigResource {
-
-  static final String BASE = "/config";
-
-  private final Storage storage;
-
-  public StyxConfigResource(Storage storage) {
-    this.storage = Objects.requireNonNull(storage);
-  }
-
-  public Stream<Route<AsyncHandler<Response<ByteString>>>> routes() {
-    final List<Route<AsyncHandler<Response<ByteString>>>> routes = Arrays.asList(
-        Route.with(
-            json(), "GET", BASE,
-            rc -> styxConfig()),
-        Route.with(
-            json(), "PATCH", BASE,
-            rc -> patchStyxConfig(rc.request()))
-    );
-
-    return Api.prefixRoutes(routes, V3);
-  }
-
-  private Response<StyxConfig> styxConfig() {
-    final boolean enabled;
-    try {
-      enabled = storage.globalEnabled();
-    } catch (IOException e) {
-      throw Throwables.propagate(e);
-    }
-    return Response.forPayload(StyxConfig.create(enabled));
-  }
-
-  private Response<StyxConfig> patchStyxConfig(Request request) {
```

Comment:
I think this is for having a big red halt button if things go crazy. Not sure if needed or if changing in datastore through cloud console is good enough.

Reply:
I did not know about this red button, it's not documented anywhere I can find (including our internal ops docs) and it seems scary to me to have this be a "public" api that anyone can use to take down styx without authorization.

Having this red button be documented and in the cloud console datastore UI (like so many of the other runtime configuration options atm) seems like an improvement & good enough to me for now.